api/v1beta1/jujucluster_types.go

```diff
@@ -36,7 +40,141 @@ type APIEndpoint struct {
 	Port int `json:"port"`
 }
 
+type Region struct {
+	// Name is the name of the region.
+	Name string `json:"name"`
+
+	// Endpoint is the region's primary endpoint URL.
+	Endpoint string `json:"endpoint"`
+
+	// IdentityEndpoint is the region's identity endpoint URL.
+	// If the cloud/region does not have an identity-specific
+	// endpoint URL, this will be empty.
+	IdentityEndpoint string `json:"identityEndpoint,omitempty"`
+
+	// StorageEndpoint is the region's storage endpoint URL.
+	// If the cloud/region does not have a storage-specific
+	// endpoint URL, this will be empty.
+	StorageEndpoint string `json:"storageEndpoint,omitempty"`
+}
+
+type AuthType string
+
+type AuthTypes []AuthType
+
```

Comment:
This struct had to be redefined in order to add JSON tagged-fields and not use empty interface types